### PR TITLE
Feature/enable draw outline for all positionables

### PR DESCRIPTION
**Pull request by piiertho**
This enable outline drawing on all positionables.
Lines are now drawn by a child node that we add on the selection. This enable to draw lines on to of everything by setting the OutlineDrawer's z index to 4096.
This will make easier to draw lines for tile creation, which will become less painfull with lines to know where to place the sprite.

### CODE REVIEW

File: src/positionable/IsometricPositionable.cpp [hunk 1]
```diff
@@ -27,6 +26,9 @@ void IsometricPositionable::_register_methods() {
     register_property("size3d", &IsometricPositionable::setSize3D, &IsometricPositionable::getSize3D, Vector3(1, 1, 1));
     register_property("is_temporary", &IsometricPositionable::setTemporary, &IsometricPositionable::isTemporary, true);
     register_property("debug_z", &IsometricPositionable::setDebugZ, &IsometricPositionable::getDebugZ, 0);
+    register_property("slope_type", &IsometricPositionable::setSlopeType, &IsometricPositionable::getSlopeType,
```
Comment (CedNaru) [inline src/positionable/IsometricPositionable.cpp]: Adding the slope here means that a map can be a slope as well. Not very logical

**Review by CedNaru — CHANGES_REQUESTED** [review]
Maybe we should create 2 new classes to make a different between the IsoContainers like maps and the IsoElement like Tiles and Placeholders.
Slope should only be for the elements.

File: src/positionable/IsometricPositionable.cpp (hunk 1 above)
Comment (piiertho) [reply src/positionable/IsometricPositionable.cpp]: Yep, that's true.
I made a new commit to make it virtual so that IsoMap can only be none.

**Comment by piiertho** [comment]
> Maybe we should create 2 new classes to make a different between the IsoContainers like maps and the IsoElement like Tiles and Placeholders.
> Slope should only be for the elements.

I'm not sure we should add inheritance chain only for slopeType, what about just making it immutable in IsometricMap ?

**Comment by piiertho** [comment]
According to discord discussion I made an IsometricElement subclass of IsometricPositionable for Tiles and Placeholders. IsometricMap no longer have slopeType.

**Review by CedNaru — APPROVED** [review]

**Pull request merged**